src/tools/dump-tool/RowDumper.cpp

```diff
@@ -0,0 +1,185 @@
+    auto fieldNum = schema_->getNumFields();
+    for (decltype(fieldNum) i = 0; i < fieldNum; ++i) {
+        // format name
+        auto name = schema_->getFieldName(i);
+        nameStream_ << ",\t";
+        nameStream_ << name;
+
+        // format value
+        valStream_ << ",\t";
+        auto type = schema_->getFieldType(i);
+        dumpOneField(type, i);
+    }
+}
+
+
+RowFormat RowDumper::dump() {
+    dumpPrefix();
+
+    if (schema_) {
+        dumpAllFields();
+    }
+
+    LOG(INFO) << nameStream_.str();
+    LOG(INFO) << valStream_.str();
+    return RowFormat(nameStream_.str(), valStream_.str());
+}
+
+
+bool TagDumper::init(kvstore::KVIterator *iter, const GraphSpaceID &spaceId,
+    SchemaManager *schemaMngPtr) {
+    row_ = iter;
+    auto key = iter->key();
+    auto val = iter->val();
+
+    auto tagId = NebulaKeyUtils::getTagId(key);
+    reader_ = RowReader::getTagPropReader(schemaMngPtr, val, spaceId, tagId);
+    if (!reader_) {
+        return false;
+    }
+
+    schema_ = schemaMngPtr->getTagSchema(spaceId, tagId, reader_->schemaVer());
+    if (!schema_) {
+        return false;
+    }
+
+    return true;
+}
+
+
+void TagDumper::dumpPrefix() {
+    auto key = row_->key();
+
+    nameStream_ << "id";
+    auto srcId = NebulaKeyUtils::getVertexID(key);
+    valStream_ << srcId;
+}
+
+
+bool EdgeDumper::init(kvstore::KVIterator *iter, const GraphSpaceID &spaceId,
+    SchemaManager *schemaMngPtr) {
+    row_ = iter;
+    auto key = iter->key();
+    auto val = iter->val();
+    if (val.empty()) {
+        return true;
+    }
+
+    auto edgeType = NebulaKeyUtils::getEdgeType(key);
+    edgeType = edgeType < 0 ? -edgeType : edgeType;
+    reader_ = RowReader::getEdgePropReader(schemaMngPtr, val, spaceId, edgeType);
+    if (!reader_) {
+        return false;
+    }
+
+    schema_ = schemaMngPtr->getEdgeSchema(spaceId, edgeType, reader_->schemaVer());
+    if (!schema_) {
+        return false;
+    }
+
+    return true;
+}
+
+
+void EdgeDumper::dumpPrefix() {
+    auto key = row_->key();
+
+    nameStream_ << "id";
+
+    auto srcId = NebulaKeyUtils::getSrcId(key);
+    valStream_ << srcId;
+
+    valStream_ << "->";
+
+    auto dstId = NebulaKeyUtils::getDstId(key);
+    valStream_ << dstId;
+
+    valStream_ << ":";
+
+    auto edgeType = NebulaKeyUtils::getEdgeType(key);
+    valStream_ << edgeType;
+}
+
+
+std::shared_ptr<RowDumper> RowDumperFactory::createRowDumper(kvstore::KVIterator *iter,
+    const GraphSpaceID &spaceId, SchemaManager *schemaMngPtr) {
+    auto key = iter->key();
+
+    if (NebulaKeyUtils::isVertex(key)) {
+        auto ret = std::make_shared<TagDumper>();
+        if (ret->init(iter, spaceId, schemaMngPtr)) {
+            return ret;
+        }
+    } else if (NebulaKeyUtils::isEdge(key)) {
+        auto ret = std::make_shared<EdgeDumper>();
+        if (ret->init(iter, spaceId, schemaMngPtr)) {
+            return ret;
+        }
+    } else {
+        // do nothing
```

Comment:
when the key is neither vertex nor edge, it's a error ?

Reply:
maybe is a system key, anyway, maybe I need print a log, thx.